Formula/awscli.rb

```diff
@@ -16,22 +27,118 @@ class Awscli < Formula
+    sha256 "b85d0567b8666149a93172712e68920734333c0ce7e89b78b3e987f71e5ed4f9"
+  end
+
+  resource "prompt-toolkit" do
+    url "https://files.pythonhosted.org/packages/0c/37/7ad3bf3c6dbe96facf9927ddf066fdafa0f86766237cff32c3c7355d3b7c/prompt_toolkit-2.0.10.tar.gz"
+    sha256 "f15af68f66e664eaa559d4ac8a928111eebd5feda0c11738b5998045224829db"
+  end
+
+  resource "pycparser" do
+    url "https://files.pythonhosted.org/packages/0f/86/e19659527668d70be91d0369aeaa055b4eb396b0f387a4f92293a20035bd/pycparser-2.20.tar.gz"
+    sha256 "2d475327684562c3a96cc71adf7dc8c4f0565175cf86b6d7a404ff4c771f15f0"
+  end
+
+  resource "python-dateutil" do
+    url "https://files.pythonhosted.org/packages/4c/c4/13b4776ea2d76c115c1d1b84579f3764ee6d57204f6be27119f13a61d0a9/python-dateutil-2.8.2.tar.gz"
+    sha256 "0123cacc1627ae19ddf3c27a5de5bd67ee4586fbdd6440d9748f8abb483d3e86"
+  end
+
+  resource "ruamel-yaml" do
+    url "https://files.pythonhosted.org/packages/9a/ee/55cd64bbff971c181e2d9e1c13aba9a27fd4cd2bee545dbe90c44427c757/ruamel.yaml-0.15.100.tar.gz"
+    sha256 "8e42f3067a59e819935a2926e247170ed93c8f0b2ab64526f888e026854db2e4"
+  end
+
+  resource "s3transfer" do
+    url "https://files.pythonhosted.org/packages/27/90/f467e516a845cf378d85f0a51913c642e31e2570eb64b352c4dc4c6cbfc7/s3transfer-0.4.2.tar.gz"
+    sha256 "cb022f4b16551edebbb31a377d3f09600dbada7363d8c5db7976e7f47732e1b2"
+  end
+
+  resource "urllib3" do
+    url "https://files.pythonhosted.org/packages/4f/5a/597ef5911cb8919efe4d86206aa8b2658616d676a7088f0825ca08bd7cb8/urllib3-1.26.6.tar.gz"
+    sha256 "f57b4c16c62fa2760b7e3d97c35b255512fb6b59a259730f36ba32ce9f8e342f"
+  end
+
+  resource "wcwidth" do
+    url "https://files.pythonhosted.org/packages/25/9d/0acbed6e4a4be4fc99148f275488580968f44ddb5e69b8ceb53fc9df55a0/wcwidth-0.1.9.tar.gz"
+    sha256 "ee73862862a156bf77ff92b09034fc4825dd3af9cf81bc5b360668d425f3c5f1"
   end
 
   def install
     venv = virtualenv_create(libexec, "python3")
-    system libexec/"bin/pip", "install", "-v", "-r", "requirements.txt",
-                              "--ignore-installed", buildpath
-    system libexec/"bin/pip", "uninstall", "-y", "awscli"
+    remaining_resources = resources
+
+    # The `awscrt` package uses its own libcrypto.a on Linux. When building
+    # _awscrt.*.so, Homebrew's env seems to cause issues due to `openssl` flags.
+    # This causes the build to fail while running `scripts/gen-ac-index` with error:
+    # ImportError: _awscrt.cpython-39-x86_64-linux-gnu.so: undefined symbol: EVP_CIPHER_CTX_init
+    on_linux do
+      remaining_resources -= [resource("awscrt")]
+      resource("awscrt").stage do
+        with_env(CFLAGS: ENV["CFLAGS"], LDFLAGS: ENV["LDFLAGS"]) do
+          ENV.prepend "CFLAGS", "-I#{Pathname.pwd}/build/deps/install/include"
+          ENV.prepend "LDFLAGS", "-L#{Pathname.pwd}/build/deps/install/lib"
+          system libexec/"bin/python3", *Language::Python.setup_install_args(libexec)
+        end
```

Comment:
This was the way to use statically linked BoringSSL, which is what upstream only supports.

Alternatively, we could patch `awscrt` to use OpenSSL. This would not be supported upstream, so we would need to deal with any issues ourselves. The following patch allowed me to do this
```diff
--- a/setup.py
+++ b/setup.py
@@ -116,15 +116,6 @@ class AwsLib:
 # They're built along with the extension, in the order listed.
 AWS_LIBS = []
 if sys.platform != 'darwin' and sys.platform != 'win32':
-    AWS_LIBS.append(AwsLib(name='aws-lc',
-                           libname='crypto',  # We link against libcrypto.a
-                           extra_cmake_args=[
-                               # We don't need libssl.a
-                               '-DBUILD_LIBSSL=OFF',
-                               # Disable running codegen on user's machine.
-                               # Up-to-date generated code is already in repo.
-                               '-DDISABLE_PERL=ON', '-DDISABLE_GO=ON',
-                           ]))
     AWS_LIBS.append(AwsLib('s2n'))
 AWS_LIBS.append(AwsLib('aws-c-common'))
 AWS_LIBS.append(AwsLib('aws-c-cal'))
@@ -254,8 +245,8 @@ def awscrt_ext():
     else:  # unix
         # linker will prefer shared libraries over static if it can find both.
         # force linker to choose static variant by using using "-l:libcrypto.a" syntax instead of just "-lcrypto".
-        libraries = [':lib{}.a'.format(x) for x in libraries]
-        libraries += ['rt']
+        #libraries = [':lib{}.a'.format(x) for x in libraries]
+        libraries += ['crypto', 'rt']

     if distutils.ccompiler.get_default_compiler() != 'msvc':
         extra_compile_args += ['-Wextra', '-Werror', '-Wno-strict-aliasing', '-std=gnu99']
```

Linkage
```console
# ldd /home/linuxbrew/.linuxbrew/Cellar/awscli/2.2.29/libexec/lib/python3.9/site-packages/_awscrt.cpython-39-x86_64-linux-gnu.so
	libcrypto.so.1.1 => /home/linuxbrew/.linuxbrew/lib/libcrypto.so.1.1 (0x0000004001c66000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x000000400214d000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x0000004002355000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x0000004002572000)
	/lib64/ld-linux-x86-64.so.2 (0x0000004000000000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x000000400293d000)
root@a1210e61a1be:/home/linuxbrew# brew edit awscli
```

Reply:
I'd rather not patch crypto/SSL implementations, so I think this is fine. I think which crypto implementation used also has implications for FIPS compliance.